hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/assertions/matchers/TransactionSidecarRecordMatcher.java

```diff
@@ -0,0 +1,325 @@
+            final Timestamp consensusTimestamp,
+            final ContractActions contractActions,
+            final ContractStateChanges contractStateChanges,
+            final ContractBytecode bytecode) {
+        super(TransactionSidecarRecord.class);
+        this.consensusTimestamp = consensusTimestamp;
+        this.actions = contractActions;
+        this.stateChanges = contractStateChanges;
+        this.bytecode = bytecode;
+    }
+
+    /**
+     * @return {@code true} if there are expected actions
+     */
+    public boolean hasActions() {
+        return actions != null;
+    }
+
+    /**
+     * @return {@code true} if there are expected state changes
+     */
+    public boolean hasStateChanges() {
+        return stateChanges != null;
+    }
+
+    /**
+     * @return a {@link TransactionSidecarRecord} with the expected fields
+     */
+    public TransactionSidecarRecord toSidecarRecord() {
+        final TransactionSidecarRecord.Builder builder = TransactionSidecarRecord.newBuilder();
+        if (consensusTimestamp != null) {
+            builder.setConsensusTimestamp(consensusTimestamp);
+        }
+        if (actions != null) {
+            builder.setActions(actions);
+        }
+        if (stateChanges != null) {
+            builder.setStateChanges(stateChanges);
+        }
+        if (bytecode != null) {
+            builder.setBytecode(bytecode);
+        }
+        return builder.build();
+    }
+
+    /**
+     * Describe the expected values for the {@link TransactionSidecarRecord}
+     * @param description {@link Description}
+     */
+    @Override
+    public void describeTo(final Description description) {
+        description.appendValue(this.toSidecarRecord());
+    }
+
+    /**
+     * Checks if the {@link TransactionSidecarRecord} matches the expected values.
+     * @param sidecarRecord the {@link TransactionSidecarRecord} to check
+     * @param mismatch {@link Description} of the mismatch
+     * @return {@code true} if the record's properties match the expected values
+     */
+    @Override
+    public boolean matchesSafely(final TransactionSidecarRecord sidecarRecord, final Description mismatch) {
+        return matchesConsensusTimestampOf(sidecarRecord, mismatch)
+                && matchesActionsOf(sidecarRecord, mismatch)
+                && matchesStateChangesOf(sidecarRecord, mismatch)
+                && matchesBytecodeOf(sidecarRecord, mismatch);
+    }
+
+    /**
+     * @param sidecarRecord the actual {@link TransactionSidecarRecord}
+     * @return {@code true} if the consensus timestamp of the given
+     * {@link TransactionSidecarRecord} matches the expected value
+     */
+    public boolean matchesConsensusTimestampOf(final TransactionSidecarRecord sidecarRecord) {
+        final Matcher<Timestamp> matcher = timestampMatcher.apply(consensusTimestamp);
+        return consensusTimestamp == null || matcher.matches(sidecarRecord.getConsensusTimestamp());
```

Comment:
The idea is that we match only the fields which are provided. 

For example, if we want to match only the `contractActions` we set only the `contractActions` of the `TransactionSidecarRecordMatcher` and all fields which are null in the matcher won't be compared to the actual object.

But you are right about the `consensusTimestamp`, it should be a required field, since we use it as an identifier when comparing the sidecar records. Without it we cannot know which record the expected record is supposed to be matched against.